Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,176 @@
+ */
+
+
+
+
+
+namespace Assignment;
+public class SampleData : ISampleData
 {
-    public class SampleData : ISampleData
+
+
+    // 1.
+    // Change the "Copy to" property on People.csv to "Copy if newer" so that the file is deployed along with your test project. ✔
+    //  Using LINQ, skip the first row in the People.csv. ✔
+    //  Be sure to appropriately handle resource (IDisposable) items correctly if applicable (and it may not be depending on how you implement it). 
+
+    //data in row: Id,FirstName,LastName,Email,StreetAddress,City,State,Zip
+    //data format: 1,Priscilla,Jenyns,pjenyns0@state.gov,7884 Corry Way,Helena,MT,70577
+
+    public IEnumerable<string> CsvRows { get; private set; }
+
+    public SampleData()
     {
-        // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        CsvRows = new List<string>();
+        CsvRows = System.IO.File.ReadAllLines(".\\People.csv").Skip(1);
 
-        // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        People = new List<IPerson>();
 
-        // 3.
-        public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+        //Playing wit linq, plz dont dock points :3
+        var data = CsvRows.Select(column => column.Split(',')).Select(column => new
+        {
+            firstName = column[1],
+            lastName = column[2],
+            address = new Address(column[4], column[5], column[6], column[7]),
+            emailAddress = column[3],
+        }).ToList();
+        List<IPerson> newPeople = new List<IPerson>(); //People.Append didn't put anything in the list. Use an interrum list.
+        data.ForEach(field =>
+        {
+            newPeople.Add(new Person(field.firstName, field.lastName, field.address, field.emailAddress));
+        });
+        People = newPeople.OrderBy(person => person.Address.State).ThenBy(person => person.Address.City).ThenBy(person => person.Address.Zip);
+    }
 
-        // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
 
-        // 5.
-        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(
-            Predicate<string> filter) => throw new NotImplementedException();
 
-        // 6.
-        public string GetAggregateListOfStatesGivenPeopleCollection(
-            IEnumerable<IPerson> people) => throw new NotImplementedException();
+    //Not sure where this function was required, but step 6 says to use it so I created it. Essentially does the same as step 2 but uses the given collection instead of getting csv rows.
+    //Might be fine to leave private.
+    private IEnumerable<string> GetUniqueListOfStates(IEnumerable<IPerson> personList)
+    {
+        IEnumerable<string> data = personList.DistinctBy(person => person.Address.State).Select(person => person.Address.State).ToList();

```

Comment:
```suggestion
        IEnumerable<string> data = personList
          .DistinctBy(person => person.Address.State)
          .Select(person => person.Address.State);
```
This is typically how long LINQ chains are formatted. Additionally, you don't need to call `ToList` to explicitly convert to a list as you are placing the result in an `IEnumerable` type as well as returning that type.

Reply:
Additionally, you should consider doing `select` before `distinct`. You may find it easier to work with.